### ISSUE TITLE
Action Recognition accuracy doesn't get better

### ISSUE THREAD

**Issue by iumyx2612** [open-mmlab/mmaction2#1590]
Before raising a question, you may need to check the following listed items.

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. I have read the [FAQ documentation](https://mmaction2.readthedocs.io/en/latest/faq.html) but cannot get the expected help.

**Dataset**
This is the structure of my original video datasset. Each video is 2-5 seconds long, contains only 1 person and 1 action display through the whole video
```
A
|_ class 0
    |_ video1.avi
    |_ video2.avi
|_ class 1
|_ class 2
.
.
|_ class 9
```
I want to use VideoDataset class so I convert my video dataset to this structure
```
B
|_ train
    |_ train_video1.avi
    |_ train_video2.avi
|_ val
    |_ val_video1.avi
    |_ val_video2.avi
|_ train_annotations.txt
|_ val_annotations.txt
```
**Config file**
I chose TSN to do action recognition
```
from mmcv import Config
cfg = Config.fromfile("configs/recognition/tsn/tsn_r50_video_1x1x8_100e_kinetics400_rgb.py")
# modify base config 
# model settings
cfg.model.cls_head.num_classes = 10 # 10 classes
cfg.model.test_cfg = test_cfg=dict(average_clips=None) 

# dataset settings
cfg.dataset_type = "VideoDataset"
cfg.data_root = "data/Gesture/train/"
cfg.data_root_val = "data/Gesture/val/"
cfg.ann_file_train = "data/Gesture/train_annotations.txt"
cfg.ann_file_val = "data/Gesture/val_annotations.txt"
cfg.ann_file_test = None

# data pipeline settings
cfg.train_pipeline = [
    dict(type='DecordInit'),
    dict(type='SampleFrames', clip_len=1, frame_interval=1, num_clips=8),
    dict(type='DecordDecode'),
    dict(type='Resize', scale=(224, 224), keep_ratio=False),
    dict(type='Normalize', **cfg.img_norm_cfg),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs', 'label'])
]
cfg.val_pipeline = [
    dict(type='DecordInit'),
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=8,
        test_mode=True),
    dict(type='DecordDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='CenterCrop', crop_size=224),
    dict(type='Normalize', **cfg.img_norm_cfg),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]

# dataloader settings
cfg.data = dict(
    videos_per_gpu=12,
    workers_per_gpu=2,
    test_dataloader=None,
    train=dict(
        type=cfg.dataset_type,
        ann_file=cfg.ann_file_train,
        data_prefix=cfg.data_root,
        pipeline=cfg.train_pipeline
    ),
    val=dict(
        type=cfg.dataset_type,
        ann_file=cfg.ann_file_val,
        data_prefix=cfg.data_root_val,
        pipeline=cfg.val_pipeline
    ),
    test=None
)
cfg.work_dir = 'workdir/TSN_noaug'
cfg.gpu_ids = range(1)
# Set seed thus the results are more reproducible
cfg.seed = 0
# We can set the checkpoint saving interval to reduce the storage cost
cfg.checkpoint_config.interval = 5
cfg.total_epochs = 50
# Save the best
cfg.evaluation.save_best='auto'
# The flag is used to determine whether it is omnisource training
cfg.setdefault('omnisource', False)

cfg.evaluation = dict(
    interval=1, metrics=['top_k_accuracy', 'mean_class_accuracy'])
cfg.checkpoint_config = dict(interval=1) # save ckpt every epoch
# LR settings
cfg.lr_config = dict(
    policy='step',
    step=[40, 80],
    type='StepLrUpdaterHook'
)
```
My full config
```
model = dict(
    type='Recognizer2D',
    backbone=dict(
        type='ResNet',
        pretrained='torchvision://resnet50',
        depth=50,
        norm_eval=False),
    cls_head=dict(
        type='TSNHead',
        num_classes=10,
        in_channels=2048,
        spatial_type='avg',
        consensus=dict(type='AvgConsensus', dim=1),
        dropout_ratio=0.4,
        init_std=0.01),
    train_cfg=None,
    test_cfg=dict(average_clips=None))
optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
optimizer_config = dict(grad_clip=dict(max_norm=40, norm_type=2))
lr_config = dict(policy='step', step=[40, 80], type='StepLrUpdaterHook')
total_epochs = 50
checkpoint_config = dict(interval=1)
log_config = dict(interval=20, hooks=[dict(type='TextLoggerHook')])
dist_params = dict(backend='nccl')
log_level = 'INFO'
load_from = None
resume_from = None
workflow = [('train', 1)]
opencv_num_threads = 0
mp_start_method = 'fork'
dataset_type = 'VideoDataset'
data_root = 'data/Gesture/train/'
data_root_val = 'data/Gesture/val/'
ann_file_train = 'data/Gesture/train_annotations.txt'
ann_file_val = 'data/Gesture/val_annotations.txt'
ann_file_test = None
img_norm_cfg = dict(
    mean=[123.675, 116.28, 103.53], std=[58.395, 57.12, 57.375], to_bgr=False)
train_pipeline = [
    dict(type='DecordInit'),
    dict(type='SampleFrames', clip_len=1, frame_interval=1, num_clips=8),
    dict(type='DecordDecode'),
    dict(type='Resize', scale=(224, 224), keep_ratio=False),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs', 'label'])
]
val_pipeline = [
    dict(type='DecordInit'),
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=8,
        test_mode=True),
    dict(type='DecordDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='CenterCrop', crop_size=224),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
test_pipeline = [
    dict(type='DecordInit'),
    dict(
        type='SampleFrames',
        clip_len=1,
        frame_interval=1,
        num_clips=25,
        test_mode=True),
    dict(type='DecordDecode'),
    dict(type='Resize', scale=(-1, 256)),
    dict(type='TenCrop', crop_size=224),
    dict(
        type='Normalize',
        mean=[123.675, 116.28, 103.53],
        std=[58.395, 57.12, 57.375],
        to_bgr=False),
    dict(type='FormatShape', input_format='NCHW'),
    dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
    dict(type='ToTensor', keys=['imgs'])
]
data = dict(
    videos_per_gpu=12,
    workers_per_gpu=2,
    test_dataloader=None,
    train=dict(
        type='VideoDataset',
        ann_file='data/Gesture/train_annotations.txt',
        data_prefix='data/Gesture/train/',
        pipeline=[
            dict(type='DecordInit'),
            dict(
                type='SampleFrames', clip_len=1, frame_interval=1,
                num_clips=8),
            dict(type='DecordDecode'),
            dict(type='Resize', scale=(224, 224), keep_ratio=False),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_bgr=False),
            dict(type='FormatShape', input_format='NCHW'),
            dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
            dict(type='ToTensor', keys=['imgs', 'label'])
        ]),
    val=dict(
        type='VideoDataset',
        ann_file='data/Gesture/val_annotations.txt',
        data_prefix='data/Gesture/val/',
        pipeline=[
            dict(type='DecordInit'),
            dict(
                type='SampleFrames',
                clip_len=1,
                frame_interval=1,
                num_clips=8,
                test_mode=True),
            dict(type='DecordDecode'),
            dict(type='Resize', scale=(-1, 256)),
            dict(type='CenterCrop', crop_size=224),
            dict(
                type='Normalize',
                mean=[123.675, 116.28, 103.53],
                std=[58.395, 57.12, 57.375],
                to_bgr=False),
            dict(type='FormatShape', input_format='NCHW'),
            dict(type='Collect', keys=['imgs', 'label'], meta_keys=[]),
            dict(type='ToTensor', keys=['imgs'])
        ]),
    test=None)
evaluation = dict(
    interval=1, metrics=['top_k_accuracy', 'mean_class_accuracy'])
work_dir = 'workdir/TSN_noaug'
gpu_ids = range(0, 1)
seed = 0
omnisource = False
```
My training log: https://drive.google.com/file/d/197pov8wRwgKBfBPb1Ab-8p11rhVm31Qd/view?usp=sharing
Colab notebook: https://colab.research.google.com/drive/1d4_Bpw3Wf3ot5H1qMls_qeTlkrP8fQIY?usp=sharing

**Comment by Dai-Wenxun** [comment]
Hi, @iumyx2612, 
After going through your colab notebook and logfile, I have two suggestions for you: 
- Your training dataset seems to be very small, with only about 1k videos, and Kinetics-400 contains about 160k videos in total. I think it is probably the problem of your small dataset.
- The original learning rate of 0.01 is set when `batch_size = num_gpus*videos_per_gpu = 8x12 = 96`, and since your `batch_size=1x12`, I suggest you try a smaller learning rate, which may achieve better convergence . The `step = 80` in the `lr_config` will not be triggered because your `total_epochs = 50`, you can remove it.

**Comment by iumyx2612** [comment]
> Hi, @iumyx2612, After going through your colab notebook and logfile, I have two suggestions for you:
> 
> * Your training dataset seems to be very small, with only about 1k videos, and Kinetics-400 contains about 160k videos in total. I think it is probably the problem of your small dataset.
> * The original learning rate of 0.01 is set when `batch_size = num_gpus*videos_per_gpu = 8x12 = 96`, and since your `batch_size=1x12`, I suggest you try a smaller learning rate, which may achieve better convergence . The `step = 80` in the `lr_config` will not be triggered because `total_epochs = 50`, you can remove it.

Thank you so so much for your suggestion, will try it out and report the results back
